### PR TITLE
removed old call to close statements

### PR DESCRIPTION
**Pull request by nitincchauhan**

### CODE REVIEW

**Comment by nitincchauhan** [comment]
something is not right 'still' :(

Failed tests: 
  TestConnectionTimeoutRetry.testConnectionIdleFill:239 Totals connections not as expected expected same:<10> was not:<7>

File: src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java [hunk 1]
```diff
@@ -220,7 +223,6 @@ public final void close() throws SQLException
          leakTask.cancel();
 
          try {
-            closeStatements();
```
Comment (brettwooldridge) [inline src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java]: Good catch.

File: src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java [hunk 2]
```diff
@@ -198,6 +198,9 @@ private final void closeStatements()
             }
             catch (SQLException e) {
                checkException(e);
+               if (delegate == ClosedConnection.CLOSED_CONNECTION) {
+                  break; //connection closed in checkException
```
Comment (brettwooldridge) [inline src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java]: I know this is more efficient, but it is optimizing a case so rare that I really don't care about the performance.

The additional code just adds cognitive load to an already messy method.

Comment (brettwooldridge) [inline src/main/java/com/zaxxer/hikari/pool/ProxyConnection.java]: Actually, it is infrequent enough for there to be unclosed Statements, especially with the prevalence of persistence frameworks (Hibernate, JPA, Play/Slick, etc.) that I'd be just as happy adding ``delegate != ClosedConnection.CLOSED_CONNECTION`` to the *for loop* conditional.

**Pull request merged**